controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -515,35 +502,17 @@ func (r *KubeadmControlPlaneReconciler) scaleDownControlPlane(
 		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: DeleteRequeueAfter}
 	}
 
-	markedForDeletion := selectedMachines.Filter(machinefilters.HasAnnotationKey(controlplanev1.DeleteForScaleDownAnnotation))
-	if len(markedForDeletion) == 0 {
-		fd := controlPlane.FailureDomainWithMostMachines()
-		machinesInFailureDomain := selectedMachines.Filter(machinefilters.InFailureDomains(fd))
-		machineToMark := machinesInFailureDomain.Oldest()
-		if machineToMark == nil {
-			logger.Info("failed to pick control plane Machine to mark for deletion")
-			return ctrl.Result{}, errors.New("failed to pick control plane Machine to mark for deletion")
-		}
-		if err := r.markWithAnnotationKey(ctx, machineToMark, controlplanev1.DeleteForScaleDownAnnotation); err != nil {
-			return ctrl.Result{}, errors.Wrapf(err, "failed to mark machine %s for deletion", machineToMark.Name)
-		}
-		markedForDeletion.Insert(machinesInFailureDomain.Oldest())
-	}
-
-	machineToDelete := markedForDeletion.Oldest()
-	if machineToDelete == nil {
-		logger.Info("failed to pick control plane Machine to delete")
-		return ctrl.Result{}, errors.New("failed to pick control plane Machine to delete")
-	}
-
 	// Ensure etcd is healthy prior to attempting to remove the member
-	if err := r.managementCluster.TargetClusterEtcdIsHealthy(ctx, util.ObjectKey(cluster), kcp.Name); err != nil {
+	if err := r.managementCluster.TargetClusterEtcdIsHealthy(ctx, util.ObjectKey(controlPlane.Cluster), controlPlane.KCP.Name); err != nil {
 		logger.Error(err, "waiting for control plane to pass etcd health check before removing a control plane machine")
-		r.recorder.Eventf(kcp, corev1.EventTypeWarning, "ControlPlaneUnhealthy", "Waiting for control plane to pass etcd health check before removing a control plane machine: %v", err)
+		r.recorder.Eventf(controlPlane.KCP, corev1.EventTypeWarning, "ControlPlaneUnhealthy", "Waiting for control plane to pass etcd health check before removing a control plane machine: %v", err)
 		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: HealthCheckFailedRequeueAfter}
 	}
+
+	machineToDelete := controlPlane.Machines.Oldest()
+
 	// If etcd leadership is on machine that is about to be deleted, move it to the newest member available.
-	etcdLeaderCandidate := ownedMachines.Newest()
+	etcdLeaderCandidate := controlPlane.EtcdLeaderCandidate()
```

Comment:
If I'm not wrong, it is missing the "If etcd leadership is on machine that is about to be deleted" bits (it always move etcd leadership)